radix_sort.py

```diff
@@ -0,0 +1,48 @@
+import random
+
+
+def radixsort(_list):
+    slots = 10
+    max_length = False
+    holder, placement = -1, 1
+
+    for i, n in enumerate(_list):
+        if n < 0:
+            raise ValueError('only accepts postive integers')
+        _list[i] = int(n)
+
+    while not max_length:
+        max_length = True
+        buckets = [list() for _ in range(slots)]
+
+        for i in _list:
+            holder = i // placement
+            buckets[holder % slots].append(i)
+            if max_length and holder > 0:
+                max_length = False
+
+        a = 0
+        for b in range(slots):
+            buck = buckets[b]
+            for i in buck:
+                _list[a] = i
+                a += 1
+        placement *= slots
+
+
+def test_bad():
+    x = []
+    for i in range(10):
+        x.append(random.randrange(100))
+    radixsort(x)
```

Comment:
This isn't particularly bad, so you won't end up seeing the "worst case" you expect when timing this.  If you really want to explore the worst case performance, look at what factor in the incoming list has the greatest effect on the time complexity of the sort.  What is that factor?  Maximize the difference in that factor between the best and worst case.